packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java

```diff
@@ -75,15 +76,31 @@ public void onError(String code, String error) {
               });
       authenticationHelper.authenticate();
     } else if (call.method.equals("getAvailableBiometrics")) {
-      FingerprintManager fingerprintMgr =
-          registrar.activity().getSystemService(FingerprintManager.class);
       ArrayList<String> biometrics = new ArrayList<String>();
-      if (fingerprintMgr.isHardwareDetected()) {
-        if (fingerprintMgr.hasEnrolledFingerprints()) {
-          biometrics.add("fingerprint");
+      try {
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
+          FingerprintManager fingerprintMgr =
+              registrar.activity().getSystemService(FingerprintManager.class);
+          if (fingerprintMgr.isHardwareDetected()) {
+            if (fingerprintMgr.hasEnrolledFingerprints()) {
+              biometrics.add("fingerprint");
+            } else {
+              biometrics.add("undefined");
+            }
+          }
         } else {
-          biometrics.add("undefined");
+          FingerprintManagerCompat fingerprintMgr =
```

Comment:
If we use the Compat libraries here, do we need to do version check above? Why not just rely on Compat libraries altogether?